spokestack/wakeword/tflite.py

```diff
@@ -94,87 +103,46 @@ def __call__(self, context: SpeechContext, frame) -> None:
         Returns: None
 
         """
+        vad_fall = self._is_speech and not context.is_speech
+        self._is_speech = context.is_speech
         if not context.is_active:
-            self.sample(context, frame)
+            self._sample(context, frame)
 
-        if not context.is_speech:
+        if vad_fall:
             self.reset()
 
-    def sample(self, context: SpeechContext, frame) -> None:
-        """ Iterates over samples from an audio frame
-
-        Args:
-            context (SpeechContext): current state of the speech pipeline
-            frame (np.ndarray): a single frame of an audio signal
+    def _sample(self, context: SpeechContext, frame) -> None:
 
-        Returns: None
+        frame /= 2 ** 15 - 1
+        frame = np.clip(frame, -1.0, 1.0)
+        frame = np.append(frame[0], frame[1:] - self.pre_emphasis * frame[:-1])
```

Comment:
`self._prev_sample` should be prepended to the frame here, in order to maintain sample continuity from frame to frame, something like this:

```
prev_sample = frame[-1]
frame -= self.pre_emphasis * np.append(self._prev_sample, frame[:-1])
self._prev_sample = prev_sample
```